tokio-util/src/udp/frame.rs

```diff
@@ -181,24 +195,7 @@ impl<C> UdpFramed<C> {
     /// coming in as it may corrupt the stream of frames otherwise being worked
     /// with.
     pub fn get_ref(&self) -> &UdpSocket {
-        &self.socket
-    }
-
-    /// Returns a mutable reference to the underlying I/O stream wrapped by
-    /// `Framed`.
-    ///
-    /// # Note
-    ///
-    /// Care should be taken to not tamper with the underlying stream of data
-    /// coming in as it may corrupt the stream of frames otherwise being worked
-    /// with.
-    pub fn get_mut(&mut self) -> &mut UdpSocket {
-        &mut self.socket
-    }
```

Comment:
If you remove `get_mut`, it becomes a breaking change. Maybe return `&mut T` here, then change `get_ref` to return `&T` for consistency.

Edit: I just noticed the `get_mut` impl further down. Not sure what I like best tbh.

Reply:
I'm good with either, I used `.borrow()` and `.borrow_mut()` because I thought it may be a breaking change if it returned `T` and a user upgraded, all of a sudden you have to call `.borrow()` additionally on that to get a `&UdpSocket`?

Reply:
Well, any existing users would have `T = UdpSocket`, so if we return `&T`, that's an `&UdpSocket`, no?

Reply:
The default param was a recent addition that changes things, good point. Okay, I've updated it to return the type params.